examples/cpp/hello_world/HelloWorldPublisher.cpp

```diff
@@ -0,0 +1,166 @@
+    }
+
+    // Register the type
+    type_.register_type(participant_);
+
+    // Create the publisher
+    PublisherQos pub_qos = PUBLISHER_QOS_DEFAULT;
+    participant_->get_default_publisher_qos(pub_qos);
+    publisher_ = participant_->create_publisher(pub_qos, nullptr);
+    if (publisher_ == nullptr)
+    {
+        throw std::runtime_error("Publisher initialization failed");
+    }
+
+    // Create the topic
+    TopicQos topic_qos = TOPIC_QOS_DEFAULT;
+    participant_->get_default_topic_qos(topic_qos);
+    topic_ = participant_->create_topic("Hello_world_topic", "HelloWorld", topic_qos);
+    if (topic_ == nullptr)
+    {
+        throw std::runtime_error("Topic initialization failed");
+    }
+
+    // Create the data writer
+    DataWriterQos writer_qos = DATAWRITER_QOS_DEFAULT;
+    publisher_->get_default_datawriter_qos(writer_qos);
+    writer_ = publisher_->create_datawriter(topic_, writer_qos, this);
+    if (writer_ == nullptr)
+    {
+        throw std::runtime_error("DataWriter initialization failed");
+    }
+}
+
+HelloWorldPublisher::~HelloWorldPublisher()
+{
+    if (writer_ != nullptr)
+    {
+        publisher_->delete_datawriter(writer_);
+    }
+    if (publisher_ != nullptr)
+    {
+        participant_->delete_publisher(publisher_);
+    }
+    if (topic_ != nullptr)
+    {
+        participant_->delete_topic(topic_);
+    }
+    DomainParticipantFactory::get_instance()->delete_participant(participant_);
+}
+
+void HelloWorldPublisher::on_publication_matched(
+        DataWriter*,
+        const PublicationMatchedStatus& info)
+{
+    if (info.current_count_change == 1)
+    {
+        matched_ = info.total_count;
+        std::cout << "Publisher matched." << std::endl;
+    }
+    else if (info.current_count_change == -1)
+    {
+        matched_ = info.total_count;
+        std::cout << "Publisher unmatched." << std::endl;
+    }
+    else
+    {
+        std::cout << info.current_count_change
+                  << " is not a valid value for PublicationMatchedStatus current count change" << std::endl;
+    }
+}
+
+void HelloWorldPublisher::run()
+{
+    std::thread pub_thread([&]
+            {
+                while (!is_stopped())
+                {
+                    if (publish())
+                    {
+                        std::cout << "Message: '" << hello_.message() << "' with index: '" << hello_.index()
+                                  << "' SENT" << std::endl;
+                    }
+                    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+                }
+            });
+    std::cout << "Publisher running. Please press Ctrl+C to stop the Publisher at any time." << std::endl;
+    signal(SIGINT, [](int signum)
+            {
+                std::cout << "SIGINT received, stopping Publisher execution." << std::endl;
+                static_cast<void>(signum); HelloWorldPublisher::stop();
+            });
```

Comment:
We should register handles for other signals IMO
```suggestion
    signal(SIGINT, [](int /*signum*/)
            {
                std::cout << "SIGINT received, stopping Publisher execution." << std::endl;
                HelloWorldPublisher::stop();
            });
```